transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollChannel.java

```diff
@@ -48,12 +49,24 @@
         readFlag = flag;
         flags |= flag;
         this.active = active;
+        fileDescriptor = new FileDescriptor(fd);
     }
 
     protected int fd() {
         return fd;
     }
 
+    /**
+     * Returns the {@link FileDescriptor} that is used by this {@link Channel}.
+     */
+    public FileDescriptor fileDescriptor() {
```

Comment:
If we don't have a plan to make the `fd()` above public, we could rename `fd()` to `fd0()` and rename this method to `fd()` instead.